source/opt/aggressive_dead_code_elim_pass.h

```diff
@@ -81,6 +81,15 @@ class AggressiveDCEPass : public MemPass {
     if (!live_insts_.Set(inst->unique_id())) {
       worklist_.push(inst);
     }
+    if (inst->GetDebugScope().GetLexicalScope() != kNoDebugScope) {
+      auto* scope =
+          get_def_use_mgr()->GetDef(inst->GetDebugScope().GetLexicalScope());
+      AddToWorklist(scope);
+    }
+    if (inst->GetDebugInlinedAt() != kNoInlinedAt) {
+      auto* inlined_at = get_def_use_mgr()->GetDef(inst->GetDebugInlinedAt());
+      AddToWorklist(inlined_at);
+    }
```

Comment:
This is not the right place for this.  This should be in the loop where we process and instruction.  Probably somewhere around [here](https://github.com/KhronosGroup/SPIRV-Tools/blob/master/source/opt/aggressive_dead_code_elim_pass.cpp#L535).  The design is that the work list is the list of instructions that need to be processed.  We do not process it when adding it to the work list, but when we remove it from the work list.

Reply:
done